### PR TITLE
feat: cloud-dns support

### LINKED ISSUE #1013: Cloud DNS for GKE Clusters
Hi All

We are using Cloud DNS for our GKE cluster, and it's now available via the terraform provider. Do you plan to implement this within the module and if so do you have an estimated release date?

Thanks
Mark.

### PR DESCRIPTION
**Pull request by DrFaust92**
Closes #1013
Continues from #1198 (thanks [SamLR](https://github.com/SamLR))

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
@DrFaust92
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_terraform
 Failed Terraform check. More details below. 
 ```bash 
Running terraform fmt
Running terraform validate
terraform_validate .
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on cluster.tf line 401, in resource "google_container_node_pool" "pools":
│
 
 401:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./examples/acm-terraform-blog-part1/terraform
Success!
 The configuration is valid.
terraform_validate ./examples/acm-terraform-blog-part2/terraform
Success!
 The configuration is valid.
terraform_validate ./examples/acm-terraform-blog-part3/terraform
Success!
 The configuration is valid.
terraform_validate ./examples/deploy_service
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on ../../cluster.tf line 401, in resource "google_container_node_pool" "pools":
│
 
 401:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./examples/disable_client_cert
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on ../../cluster.tf line 401, in resource "google_container_node_pool" "pools":
│
 
 401:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./examples/node_pool
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on ../../modules/beta-public-cluster/cluster.tf line 559, in resource "google_container_node_pool" "pools":
│
 
 559:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./examples/node_pool_update_variant
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on ../../modules/private-cluster-update-variant/cluster.tf line 503, in resource "google_container_node_pool" "pools":
│
 
 503:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./examples/node_pool_update_variant_beta
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on ../../modules/beta-private-cluster-update-variant/cluster.tf line 667, in resource "google_container_node_pool" "pools":
│
 
 667:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./examples/node_pool_update_variant_public_beta
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on ../../modules/beta-public-cluster-update-variant/cluster.tf line 648, in resource "google_container_node_pool" "pools":
│
 
 648:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./examples/private_zonal_with_networking
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on ../../modules/private-cluster/cluster.tf line 414, in resource "google_container_node_pool" "pools":
│
 
 414:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./examples/regional_private_node_pool_oauth_scopes
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on ../../modules/private-cluster/cluster.tf line 414, in resource "google_container_node_pool" "pools":
│
 
 414:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./examples/safer_cluster
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on ../../modules/beta-private-cluster/cluster.tf line 578, in resource "google_container_node_pool" "pools":
│
 
 578:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./examples/safer_cluster_iap_bastion
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on ../../modules/beta-private-cluster/cluster.tf line 578, in resource "google_container_node_pool" "pools":
│
 
 578:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./examples/shared_vpc
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on ../../cluster.tf line 401, in resource "google_container_node_pool" "pools":
│
 
 401:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./examples/simple_autopilot_private
Success!
 The configuration is valid.
terraform_validate ./examples/simple_autopilot_public
Success!
 The configuration is valid.
terraform_validate ./examples/simple_regional
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on ../../cluster.tf line 401, in resource "google_container_node_pool" "pools":
│
 
 401:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./examples/simple_regional_beta
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on ../../modules/beta-public-cluster/cluster.tf line 559, in resource "google_container_node_pool" "pools":
│
 
 559:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./examples/simple_regional_private
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on ../../modules/private-cluster/cluster.tf line 414, in resource "google_container_node_pool" "pools":
│
 
 414:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./examples/simple_regional_private_beta
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on ../../modules/beta-private-cluster/cluster.tf line 578, in resource "google_container_node_pool" "pools":
│
 
 578:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./examples/simple_regional_with_kubeconfig
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on ../../cluster.tf line 401, in resource "google_container_node_pool" "pools":
│
 
 401:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./examples/simple_regional_with_networking
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on ../../cluster.tf line 401, in resource "google_container_node_pool" "pools":
│
 
 401:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./examples/simple_zonal_private
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on ../../modules/private-cluster/cluster.tf line 414, in resource "google_container_node_pool" "pools":
│
 
 414:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./examples/simple_zonal_with_acm
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on ../../cluster.tf line 401, in resource "google_container_node_pool" "pools":
│
 
 401:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./examples/simple_zonal_with_asm
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on ../../cluster.tf line 401, in resource "google_container_node_pool" "pools":
│
 
 401:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./examples/simple_zonal_with_hub
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on ../../cluster.tf line 401, in resource "google_container_node_pool" "pools":
│
 
 401:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./examples/simple_zonal_with_hub_kubeconfig
Success!
 The configuration is valid.
terraform_validate ./examples/stub_domains
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on ../../cluster.tf line 401, in resource "google_container_node_pool" "pools":
│
 
 401:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./examples/stub_domains_private
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on ../../modules/private-cluster/cluster.tf line 414, in resource "google_container_node_pool" "pools":
│
 
 414:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./examples/stub_domains_upstream_nameservers
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on ../../cluster.tf line 401, in resource "google_container_node_pool" "pools":
│
 
 401:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./examples/upstream_nameservers
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on ../../cluster.tf line 401, in resource "google_container_node_pool" "pools":
│
 
 401:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./examples/workload_identity
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on ../../cluster.tf line 401, in resource "google_container_node_pool" "pools":
│
 
 401:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./examples/workload_metadata_config
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on ../../modules/private-cluster/cluster.tf line 414, in resource "google_container_node_pool" "pools":
│
 
 414:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./modules/acm
Success!
 The configuration is valid.
terraform_validate ./modules/asm
Success!
 The configuration is valid.
terraform_validate ./modules/auth
Success!
 The configuration is valid.
terraform_validate ./modules/beta-autopilot-private-cluster
Success!
 The configuration is valid.
terraform_validate ./modules/beta-autopilot-public-cluster
Success!
 The configuration is valid.
terraform_validate ./modules/beta-private-cluster
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on cluster.tf line 578, in resource "google_container_node_pool" "pools":
│
 
 578:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./modules/beta-private-cluster-update-variant
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on cluster.tf line 667, in resource "google_container_node_pool" "pools":
│
 
 667:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./modules/beta-public-cluster
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on cluster.tf line 559, in resource "google_container_node_pool" "pools":
│
 
 559:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./modules/beta-public-cluster-update-variant
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on cluster.tf line 648, in resource "google_container_node_pool" "pools":
│
 
 648:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./modules/binary-authorization
Success!
 The configuration is valid.
terraform_validate ./modules/fleet-membership
Success!
 The configuration is valid.
terraform_validate ./modules/hub-legacy
Success!
 The configuration is valid.
terraform_validate ./modules/private-cluster
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on cluster.tf line 414, in resource "google_container_node_pool" "pools":
│
 
 414:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./modules/private-cluster-update-variant
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on cluster.tf line 503, in resource "google_container_node_pool" "pools":
│
 
 503:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./modules/safer-cluster
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on ../beta-private-cluster/cluster.tf line 578, in resource "google_container_node_pool" "pools":
│
 
 578:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./modules/safer-cluster-update-variant
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on ../beta-private-cluster-update-variant/cluster.tf line 667, in resource "google_container_node_pool" "pools":
│
 
 667:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./modules/services
Success!
 The configuration is valid.
terraform_validate ./modules/workload-identity
Success!
 The configuration is valid.
terraform_validate ./test/fixtures/beta_cluster
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on ../../../modules/beta-public-cluster/cluster.tf line 559, in resource "google_container_node_pool" "pools":
│
 
 559:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./test/fixtures/deploy_service
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on ../../../cluster.tf line 401, in resource "google_container_node_pool" "pools":
│
 
 401:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./test/fixtures/disable_client_cert
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on ../../../cluster.tf line 401, in resource "google_container_node_pool" "pools":
│
 
 401:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./test/fixtures/node_pool
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on ../../../modules/beta-public-cluster/cluster.tf line 559, in resource "google_container_node_pool" "pools":
│
 
 559:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./test/fixtures/node_pool_update_variant
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on ../../../modules/private-cluster-update-variant/cluster.tf line 503, in resource "google_container_node_pool" "pools":
│
 
 503:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./test/fixtures/private_zonal_with_networking
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on ../../../modules/private-cluster/cluster.tf line 414, in resource "google_container_node_pool" "pools":
│
 
 414:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./test/fixtures/safer_cluster
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on ../../../modules/beta-private-cluster/cluster.tf line 578, in resource "google_container_node_pool" "pools":
│
 
 578:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./test/fixtures/safer_cluster_iap_bastion
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on ../../../modules/beta-private-cluster/cluster.tf line 578, in resource "google_container_node_pool" "pools":
│
 
 578:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./test/fixtures/sandbox_enabled
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on ../../../modules/beta-public-cluster/cluster.tf line 559, in resource "google_container_node_pool" "pools":
│
 
 559:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./test/fixtures/simple_autopilot_private
Success!
 The configuration is valid.
terraform_validate ./test/fixtures/simple_autopilot_public
Success!
 The configuration is valid.
terraform_validate ./test/fixtures/simple_regional
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on ../../../cluster.tf line 401, in resource "google_container_node_pool" "pools":
│
 
 401:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./test/fixtures/simple_regional_private
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on ../../../modules/private-cluster/cluster.tf line 414, in resource "google_container_node_pool" "pools":
│
 
 414:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./test/fixtures/simple_regional_with_kubeconfig
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on ../../../cluster.tf line 401, in resource "google_container_node_pool" "pools":
│
 
 401:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./test/fixtures/simple_regional_with_networking
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on ../../../cluster.tf line 401, in resource "google_container_node_pool" "pools":
│
 
 401:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./test/fixtures/simple_zonal
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on ../../../cluster.tf line 401, in resource "google_container_node_pool" "pools":
│
 
 401:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./test/fixtures/simple_zonal_private
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on ../../../modules/private-cluster/cluster.tf line 414, in resource "google_container_node_pool" "pools":
│
 
 414:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./test/fixtures/simple_zonal_with_asm
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on ../../../cluster.tf line 401, in resource "google_container_node_pool" "pools":
│
 
 401:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./test/fixtures/stub_domains
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on ../../../cluster.tf line 401, in resource "google_container_node_pool" "pools":
│
 
 401:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./test/fixtures/stub_domains_private
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on ../../../modules/private-cluster/cluster.tf line 414, in resource "google_container_node_pool" "pools":
│
 
 414:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./test/fixtures/stub_domains_upstream_nameservers
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on ../../../cluster.tf line 401, in resource "google_container_node_pool" "pools":
│
 
 401:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./test/fixtures/upstream_nameservers
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on ../../../cluster.tf line 401, in resource "google_container_node_pool" "pools":
│
 
 401:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./test/fixtures/workload_identity
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on ../../../cluster.tf line 401, in resource "google_container_node_pool" "pools":
│
 
 401:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./test/fixtures/workload_metadata_config
╷
│
 
Error: 
Unsupported block type
│
 
│
 
  on ../../../modules/private-cluster/cluster.tf line 414, in resource "google_container_node_pool" "pools":
│
 
 414:   dynamic 
"dns_config"
 {
│
 
│
 
Blocks of type "dns_config" are not expected here.
╵
terraform_validate ./test/setup
Success!
 The configuration is valid.
 ```

**Comment by comment-bot-dev** [comment]
@DrFaust92
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_whitespace
 Failed whitespace check. More details below. 
 ```bash 
Checking for trailing whitespace
./modules/beta-autopilot-public-cluster/cluster.tf:151:  }  
./modules/beta-autopilot-private-cluster/cluster.tf:151:  }  
Error: Trailing whitespace found in the lines above.
Checking for missing newline at end of file
 ``` 
- ⚠️check_terraform
 Failed Terraform check. More details below. 
 ```bash 
Running terraform fmt
modules/beta-autopilot-public-cluster/cluster.tf
--- old/modules/beta-autopilot-public-cluster/cluster.tf
+++ new/modules/beta-autopilot-public-cluster/cluster.tf
@@ -148,7 +148,7 @@
       cluster_dns_scope  = var.cluster_dns_scope
       cluster_dns_domain = var.cluster_dns_domain
     }
-  }  
+  }
 
   timeouts {
     create = lookup(var.timeouts, "create", "45m")
Error: terraform fmt failed with exit code 3
Check the output for diffs and correct using terraform fmt <dir>
modules/beta-autopilot-private-cluster/cluster.tf
--- old/modules/beta-autopilot-private-cluster/cluster.tf
+++ new/modules/beta-autopilot-private-cluster/cluster.tf
@@ -148,7 +148,7 @@
       cluster_dns_scope  = var.cluster_dns_scope
       cluster_dns_domain = var.cluster_dns_domain
     }
-  }  
+  }
 
   timeouts {
     create = lookup(var.timeouts, "create", "45m")
Error: terraform fmt failed with exit code 3
Check the output for diffs and correct using terraform fmt <dir>
 ```

**Comment by comment-bot-dev** [comment]
@DrFaust92
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@DrFaust92
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@DrFaust92
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@DrFaust92
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@DrFaust92
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Review by bharathkkb — CHANGES_REQUESTED** [review]
Thanks for the PR @DrFaust92 
One question

**Comment by comment-bot-dev** [comment]
@DrFaust92
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

File: modules/beta-autopilot-private-cluster/cluster.tf [hunk 1]
```diff
@@ -141,6 +141,15 @@ resource "google_container_cluster" "primary" {
   }
 
 
+  dynamic "dns_config" {
```
[the remark this answers is not in the document]
Comment (DrFaust92) [reply modules/beta-autopilot-private-cluster/cluster.tf]: added if for autopilot

**Review by bharathkkb — APPROVED** [review]

**Pull request merged**